src/Core/Services/TokenService.cs

```diff
@@ -209,6 +209,10 @@ public async Task<bool> GetIsExternal()
             {
                 await GetTokenAsync();
             }
+            if (_accessTokenForDecoding == null)
```

Comment:
Would it make more sense to change this to a `while` instead of `if` with the call to `await GetTokenAsync()`, perhaps with a cancellation token or counter within that loop? What impacts does `GetIsExternal()` have on the UI/UX if it returns `false` (since this is a race condition vs. a valid `null` condition)?

Reply:
From what I can see returning false prevents the `RemoveMasterPasswordPage` from launching.  As for looping vs returning false - I've gone through what feels like hundreds of flow possibilities and I keep disproving my own theories, so the only thing I'm sure of is that I don't know (we've never been able to repro this in-house). I'm counting on subsequent calls to `TabsPage.OnAppearing` to trigger the page launch once the token becomes available, but looping with a counter _could_ be a better approach.  Need input from @jlf0dev on this one.

Reply:
I'm not sure how we could improve this either, besides maybe loading the token a few lines sooner? @mp-bw we should be filling the token from the `LogInHelper` but is it not getting there in time?

That being said, I _think_ I like it more without a loop. The delay in relying on subsequent calls to `TabsPage.OnAppearing` to check for the token feels safer than the danger of hanging the UI while we wait.

Reply:
> we should be filling the token from the `LogInHelper` but is it not getting there in time?

It _should_ be there unless we intentionally don't store it based on configuration (combination of logout on timeout and no biometrics I believe prevent token storage).  ~~Perhaps in that state, the app is backgrounded on timeout, brought to the foreground and firing `OnAppearing` before the vaultTimeoutService re-routes to the login screen.  Though If that were the case I would expect to see a higher number of crashes per user.~~ Except a "resume" flow like that would still have the token in memory, so add that to the dead theory pile.